GDJS/Runtime/runtimeobject.ts

```diff
@@ -1617,40 +1618,128 @@ namespace gdjs {
       objectsLists: ObjectsLists,
       ignoreTouchingEdges: boolean
     ): boolean {
-      let moved = false;
-      let xMove = 0;
-      let yMove = 0;
+      let moveXArray: Array<float> =
+        RuntimeObject.separateFromObjectsStatics.moveXArray;
+      let moveYArray: Array<float> =
+        RuntimeObject.separateFromObjectsStatics.moveYArray;
+      moveXArray.length = 0;
+      moveYArray.length = 0;
+
       const hitBoxes = this.getHitBoxes();
+
       for (const name in objectsLists.items) {
         if (objectsLists.items.hasOwnProperty(name)) {
-          const objects = objectsLists.items[name];
-
-          //Check if their is a collision with each object
-          for (let i = 0, len = objects.length; i < len; ++i) {
-            if (objects[i].id != this.id) {
-              const otherHitBoxes = objects[i].getHitBoxes();
-              for (let k = 0, lenk = hitBoxes.length; k < lenk; ++k) {
-                for (let l = 0, lenl = otherHitBoxes.length; l < lenl; ++l) {
-                  const result = gdjs.Polygon.collisionTest(
-                    hitBoxes[k],
-                    otherHitBoxes[l],
-                    ignoreTouchingEdges
-                  );
-                  if (result.collision) {
-                    xMove += result.move_axis[0];
-                    yMove += result.move_axis[1];
-                    moved = true;
-                  }
+          const otherObjects = objectsLists.items[name];
+
+          // Check if their is a collision with each object
+          for (const otherObject of otherObjects) {
+            if (otherObject.id === this.id) {
+              continue;
+            }
+            const otherHitBoxes = otherObject.getHitBoxes();
+            for (const hitBox of hitBoxes) {
+              for (const otherHitBox of otherHitBoxes) {
+                const result = gdjs.Polygon.collisionTest(
+                  hitBox,
+                  otherHitBox,
+                  ignoreTouchingEdges
+                );
+                if (result.collision) {
+                  moveXArray.push(result.move_axis[0]);
+                  moveYArray.push(result.move_axis[1]);
                 }
               }
             }
           }
         }
       }
+      return this._doSeparateFromObjects(moveXArray, moveYArray);
+    }
+
+    /**
+     * Separate the object from others objects, using the results from collisionTest call.
+     * @param moveXArray
+     * @param moveYArray
+     * @return true if the object was moved
+     */
+    private _doSeparateFromObjects(
```

Comment:
Maybe we could find a slightly different names, like "_moveFollowingSeparatingVectors"?

Also, nitpicking: I see this method is only accessing `this.xxx` public methods. I wonder if we could then extract it outside of the class (passing the object as parameter)?
This just helps clarify that this is a "helper" for the two other functions, and make sure this is not used outside of the class.